File: FlashWizBE/FlashWizServer/src/main/java/com/flashwizserver/repository/FolderRepository.java
package com.flashwizserver.repository;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.*;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import com.flashwizserver.model.Folder;

@Repository

public interface FolderRepository extends CrudRepository<Folder, Integer> {
	public Long countById(Integer id);
	  @Query("SELECT f FROM Folder f WHERE f.user.id = :userId")
	    List<Folder> findByUserId(@Param("userId") Integer userId);
}

File: FlashWiz_FrontEnd/app/src/main/java/com/example/flashwiz_fe/domain/repository/FolderRepository.kt
package com.example.flashwiz_fe.domain.repository

import com.example.flashwiz_fe.domain.model.FolderDetail

interface FolderRepository {
    suspend fun getFoldersByUserId(userId: Int?) : List<FolderDetail>

    suspend fun getFolderById(folderId: Int) : FolderDetail
}